### ISSUE TITLE
braintrust push tool.ts is broken

### ISSUE THREAD

**Issue by ivanvanderbyl** [braintrustdata/braintrust-sdk#390]
Hi team,

Trying out the example from https://www.braintrust.dev/docs/guides/prompts#using-tools produces the following error:

```
braintrust push tools/github.ts
Processing 1 file...

TypeError: Cannot read properties of undefined (reading 'Latest')
```

I believe it has something to do with a missing null / undefined check for the result from loading the typescript module.

**Update**

The issue seems to be related to https://github.com/microsoft/TypeScript/issues/56903 and used here https://github.com/braintrustdata/braintrust-sdk/blob/52cda4d77c07f76113b392291e947e02010a8ef8/js/src/functions/infer-source.ts#L176-L190

**Comment by ankrgyl** [comment]
Interesting, thanks for flagging. I would love to reproduce this if possible. What version of typescript are you running? Anything else that's unique or special about your environment? If it's easy to share a repro, even better!

**Comment by ivanvanderbyl** [comment]
Hi @ankrgyl  I was able to resolve by manually editing the source of the compiled module to use `require("typescript")` instead of `await import` as per the bug listed above, which resolved the issue.

**Comment by ivanvanderbyl** [comment]
I'm using TS 5.5.4

**Comment by ankrgyl** [comment]
FYI this fix is live in 0.0.163
